packages/web/components/swap-tool/trade-details.tsx

```diff
@@ -0,0 +1,511 @@
+import { Disclosure } from "@headlessui/react";
+import { Dec, IntPretty, PricePretty, RatePretty } from "@keplr-wallet/unit";
+import { EmptyAmountError } from "@osmosis-labs/keplr-hooks";
+import { DEFAULT_VS_CURRENCY } from "@osmosis-labs/server";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import { useEffect, useMemo, useState } from "react";
+import { useMeasure } from "react-use";
+
+import { Icon } from "~/components/assets/icon";
+import { SkeletonLoader, Spinner } from "~/components/loaders";
+import { RouteLane } from "~/components/swap-tool/split-route";
+import { GenericDisclaimer } from "~/components/tooltip/generic-disclaimer";
+import { RecapRow } from "~/components/ui/recap-row";
+import {
+  useDisclosure,
+  UseDisclosureReturn,
+  usePreviousWhen,
+  useSlippageConfig,
+  useTranslation,
+} from "~/hooks";
+import { useSwap } from "~/hooks/use-swap";
+import { formatPretty, getPriceExtendedFormatOptions } from "~/utils/formatter";
+import { RouterOutputs } from "~/utils/trpc";
+
+interface TradeDetailsProps {
+  swapState: ReturnType<typeof useSwap>;
+  slippageConfig: ReturnType<typeof useSlippageConfig>;
+  type: "limit" | "market";
+  outAmountLessSlippage?: IntPretty;
+  outFiatAmountLessSlippage?: PricePretty;
+  inPriceFetching?: boolean;
+  treatAsStable?: string;
+  makerFee?: Dec;
+  tab?: "buy" | "sell";
+}
+
+export const TradeDetails = observer(
+  ({
+    swapState,
+    inPriceFetching,
+    treatAsStable,
+    type,
+    makerFee,
+    tab,
+  }: Partial<TradeDetailsProps>) => {
+    const { t } = useTranslation();
+    const routesVisDisclosure = useDisclosure();
+
+    const [outAsBase, setOutAsBase] = useState(!tab || tab === "buy");
+
+    const [details, { height: detailsHeight }] = useMeasure<HTMLDivElement>();
+
+    const isInAmountEmpty = useMemo(
+      () => swapState?.inAmountInput.error instanceof EmptyAmountError,
+      [swapState?.inAmountInput.error]
+    );
+
+    const isLoading = useMemo(
```

Comment:
As stated elsewhere, we can simplify these boolean memos by removing the useMemo hook. Performance wise, the reason there's no net benefit there is a boolean check will have to take place on the dependency array anyways, and if a reference changes then another boolean check will have to take place within the memo itself.